daemon/graphdriver/windows/windows.go

```diff
@@ -140,22 +140,30 @@ func (d *Driver) Create(id, parent string) error {
 	if _, err := os.Lstat(d.dir(parent)); err == nil {
 		if err := d.setLayerChain(id, layerChain); err != nil {
 			if err2 := hcsshim.DestroyLayer(d.info, id); err2 != nil {
-				logrus.Warnf("Failed to DestroyLayer %s: %s", id, err)
+				logrus.Warnf("Failed to DestroyLayer %s: %s", id, err2)
 			}
 			return err
 		}
 	} else if os.IsNotExist(err) {
-		// If the parent doesn't exist, this must be a special creation for an image
-		// registered at an alternate location. Use the parent id as the alternate ID.
+		if !d.Exists(parent) {
+			// The graphdriver did not find the parent in the central store.
+			if err2 := hcsshim.DestroyLayer(d.info, id); err2 != nil {
+				logrus.Warnf("Failed to DestroyLayer %s: %s", id, err2)
+			}
+			return fmt.Errorf("Cannot create layer with missing parent %s: %s", parent, err)
+		}
+		// If the parent doesn't exist but the graphdriver found it, this must be a
+		// special creation for an image registered at an alternate location.
+		// Use the parent id as the alternate ID.
 		if err := d.setId(id, parent); err != nil {
 			if err2 := hcsshim.DestroyLayer(d.info, id); err2 != nil {
-				logrus.Warnf("Failed to DestroyLayer %s: %s", id, err)
+				logrus.Warnf("Failed to DestroyLayer %s: %s", id, err2)
 			}
 			return err
 		}
```

Comment:
I'm not sure I understand what condition you are talking about.  If the error is a NotExist, then the code will check if the driver is aware of it.  If it also doesn't exist in the driver, we return an error, but if it does exist in the driver then we try to set the ID.  Which case is the one that is problematic?